app/controllers/entries_controller.rb

```diff
@@ -0,0 +1,72 @@
+class EntriesController < ApplicationController
+  before_action :authenticate_user!, only: %i[ new create edit update destroy ]
+  before_action :set_entry, only: %i[ edit update destroy ]
+
+  def index
+    @entries = Entry.all
+  end
+
+  def show
+    @entry = Entry.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    redirect_to entries_path
+  end
+
+  def new
+    @entry = Entry.new
+  end
+
+  def edit
+  end
+
+  def create
+    @entry = Entry.new(entry_params)
+    @entry.team = current_user.team
+
+    respond_to do |format|
+      if @entry.save
+        format.html { redirect_to entry_url(@entry), notice: "Entry was successfully created." }
+        format.json { render :show, status: :created, location: @entry }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @entry.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def update
+    respond_to do |format|
+      if @entry.update(entry_params)
+        format.html { redirect_to entry_url(@entry), notice: "Entry was successfully updated." }
+        format.json { render :show, status: :ok, location: @entry }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @entry.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    respond_to do |format|
+      if current_user.team == @entry.team
```

Comment:
I think you can remove this since it's already being scoped in `set_entry` right?